ERCS/erc-7588.md

```diff
@@ -0,0 +1,102 @@
+created: 2024-01-01
+requires: 4844
+---
+
+## Abstract
+
+This EIP proposes a standard for attaching metadata to blobs carried by shard blob transactions, as introduced by [EIP-4844](./eip-4844.md). The metadata is a JSON object that conforms to a predefined schema and is encoded in hexadecimal with UTF-8 character encoding. The metadata is put in the `data` field of the blob transaction.
+
+## Motivation
+
+[EIP-4844](./eip-4844.md) defines a new type of transaction, called “blob transaction”, which contains a list of blobs and their KZG commitments and proofs. Blob transactions are designed to be used by rollups to post their layer 2 transaction data to Ethereum layer 1.
+
+Typically, rollups keep track of their own posted blob transactions, download blobs and decode their layer 2 transaction data from blobs on demand. However, some third party solutions, such as Portal Network, blobscan, etc., may also index all the blobs ever posted to Ethereum and provide querying services of them. It would greatly improve the visibility and auditability of blobs if we could attach some metadata to the blobs, such as `originator`, `description`, `content_type`, etc.
+
+Moreover, some `dStorage` projects may use blob transactions to post user data to Ethereum, sync and permanently store the blobs off-chain for future retrieval. Attaching metadata to blobs would also inspire novel applications, such as inscriptions, etc.
+
+
+## Specification
+
+### Metadata JSON Schema
+
+The metadata is a JSON object that conforms to the following schema:
+
+```json
+{
+    "title": "Blobs Metadata",
+    "type": "object",
+    "properties": {
+        "originator": {
+            "type": "string",
+            "description": "Identifies the originator of the carried blobs"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the contents of the blobs"
+        },
+        "content_type": {
+            "type": "string",
+            "description": "Describes the mime type of the blobs"
+        },
+        "extras": {
+            "type": "string",
+            "description": "Dynamic extra information of the blobs"
+        },
+        "blobs": {
+            "type": "array",
+            "items": {
+                "description": {
+                    "type": "string",
+                    "description": "Describes the content of the i'th blob"
+                },
+                "content_type": {
+                    "type": "string",
+                    "description": "Describes the mime type of the i'th blob"
+                },
+                "extras": {
+                    "type": "string",
+                    "description": "Dynamic extra information of the i'th blob"
+                },
+            }
+        }
+    }
+}
+```
+
+
+### Metadata Encoding
+
+The metadata JSON object should be converted to hexadecimal with UTF-8 character encoding, and put in the `data` field of the blob transaction.
```

Comment:
To me, this reads as first encode as hexadecimal then put that on chain. So for an input of `{}`, the calldata would end up being `0x37623762`. That doesn't really make sense to me. Please make this section less ambiguous.